udata_tabular_preview/views.py

```diff
@@ -32,6 +34,19 @@ def load_explore_script(ctx):
     return render_template('script.html')
 
 
+@template_hook('dataset.display.explore-button', when=can_explore_dataset)
+def load_explore_button(ctx):
+    dataset = ctx.get('dataset', None)
+    url = ""
+    explore_url = current_app.config.get('TABULAR_EXPLORE_URL')
+    netloc = urlsplit(explore_url).netloc
+    for resource in dataset.resources:
+        if can_explore(resource):
+            url = resource.preview_url
+            break
```

Comment:
A shorter way to get the first `url`. It would raise a 500 if none is found though instead of the empty url fallback.

```suggestion
    explore_url = current_app.config.get('TABULAR_EXPLORE_URL')
    netloc = urlsplit(explore_url).netloc
    url = next(res.preview_url for res in dataset.resources if can_explore(res))
```

Reply:
I think the `500` case can't happen because the function is called after `can_explore_dataset` that also use `can_preview`.